File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int i = 0;
        int j = 0;
        while(i < str1.length() && j < str2.length()){
            char c1 = str1.charAt(i);
            char c2 = str2.charAt(j);
            int diff = c2-c1;
            if(c1 == c2 || diff == 1 || diff == -25) j += 1;
            i += 1;
        }
        return (j == str2.length())?true:false;
    }
}

File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.py
class Solution:
    def canMakeSubsequence(self, str1: str, str2: str) -> bool:
        i = 0
        j = 0
        while i < len(str1) and j < len(str2):
            diff = ord(str2[j])-ord(str1[i]) 
            if str1[i] == str2[j] or diff == 1 or diff == -25:
                j += 1
            i += 1
        if j == len(str2):
            return True
        else:
            return False
